cmd/qat_plugin/dpdkdrv/dpdkdrv.go

```diff
@@ -87,19 +132,54 @@ func NewDevicePlugin(maxDevices int, kernelVfDrivers string, dpdkDriver string)
 		}
 	}
 
-	return newDevicePlugin(pciDriverDirectory, pciDeviceDirectory, maxDevices, kernelDrivers, dpdkDriver), nil
+	allocationPolicyFunc := getAllocationPolicy(preferredAllocationPolicy)
+	if allocationPolicyFunc == nil {
+		return nil, errors.Errorf("wrong allocation policy: %s", preferredAllocationPolicy)
+	}
+
+	return newDevicePlugin(pciDriverDirectory, pciDeviceDirectory, maxDevices, kernelDrivers, dpdkDriver, allocationPolicyFunc), nil
+}
+
+//getAllocationPolicy returns a func that fits the policy given as a parameter. It returns nonePolicy when the flag is not set, and it returns nil when the policy is not valid value.
+func getAllocationPolicy(preferredAllocationPolicy string) preferredAllocationPolicyFunc {
+	switch {
+	case !isFlagSet("allocation-policy"):
+		return nonePolicy
+	case preferredAllocationPolicy == "packed":
+		return packedPolicy
+	case preferredAllocationPolicy == "balanced":
+		return balancedPolicy
+	default:
+		return nil
+	}
+}
+
+// isFlagSet returns true when the flag that has the same name as the parameter is set.
+func isFlagSet(name string) bool {
+	set := false
+
+	flag.Visit(func(f *flag.Flag) {
+		if f.Name == name {
+			set = true
+		}
+	})
+
+	return set
 }
 
-func newDevicePlugin(pciDriverDir, pciDeviceDir string, maxDevices int, kernelVfDrivers []string, dpdkDriver string) *DevicePlugin {
-	return &DevicePlugin{
+func newDevicePlugin(pciDriverDir, pciDeviceDir string, maxDevices int, kernelVfDrivers []string, dpdkDriver string, preferredAllocationPolicyFunc preferredAllocationPolicyFunc) *DevicePlugin {
+	dp := &DevicePlugin{
```

Comment:
this can be kept as:
```suggestion
	return  &DevicePlugin{
```

Reply:
Thanks.! I changed!